R/tspa.R

```diff
@@ -113,27 +114,50 @@
 
 
 tspa <- function(model, data, reliability = NULL, se = NULL,
-                 vc = NULL, cross_loadings = NULL, ...) {
+                 fsT = NULL, fsL = NULL, ...) {
+
+  if (nchar(model) == 0) {
+    stop("Please provide a structural path model.")
+  }
+
   if (!is.null(reliability)) {
     stop("tspa() currently does not support reliability model")
   }
+
   if (!is.data.frame(se)) {
     se <- as.data.frame(as.list(se))
   }
-  multigroup <- nrow(se) == 1 | is.list(vc)
+  multigroup <- nrow(se) > 1 | is.list(fsT)
+
+  if (sum(is.null(fsT), is.null(fsL)) == 1) {
+    stop("Please provide both or none of fsT and fsL.")
+  }
+
+  if (!is.null(fsT)) {
+    fs_names <- ifelse(multigroup, colnames(fsT[[1]]), colnames(fsT))
+    dat_names <- ifelse(multigroup, names(data[[1]]), names(data))
+    names_match <- lapply(fs_names, function(x) x %in% dat_names) |> unlist()
+    if (any(!names_match)) {
+      stop("Names of factor score variables do not match those in the input data.")
+    }
+  }
 
   if (multigroup) {
```

Comment:
Just checking: do we now use the multigroup function for everything?

Reply:
That's a good question. Yes but we haven't merged the changes in the `growth` branch to `devel`. I forsee there will be quite some conflicts to resolve and plan to complete this pull request before moving on to the pull request from `growth`.

Reply:
I see. Let's wait for that then.